cdap-formats/src/test/java/io/cdap/cdap/format/StructuredRecordBuilderTest.java

```diff
@@ -216,6 +217,16 @@ public void testDecimalLogicalType() {
     Assert.assertNull(record.getDecimal("d"));
   }
 
+  @Test
+  public void testDecimalLogicalFromByteBuffer() {
+    Schema schema = Schema.recordOf("test", Schema.Field.of("d", Schema.nullableOf(Schema.decimalOf(3, 2))));
+    BigDecimal d = new BigDecimal(new BigInteger("111"), 2);
+    StructuredRecord record = StructuredRecord.builder(schema)
+      .set("d", ByteBuffer.wrap(d.unscaledValue().toByteArray())).build();
```

Comment:
The structured record is being created from the json string in `StructuredRecordStringConverter#fromJsonString`. Would it be better to add the test in `StructuredRecordStringConverterTest`?

Reply:
Moved to the `StructuredRecordStringConverterTest`.